### ISSUE TITLE
query iceberg table very slow

### ISSUE THREAD

**Issue by rlcyf** [trinodb/trino#7336]
iceberg 0.11.0
trino 351

trino worker*3
16CORE 32G RAM

```
trino:odsx1> select count(*) from sample03;
 _col0 
-------
     9 
(1 row)

Query 20210318_152248_00001_5mzin, FINISHED, 3 nodes
Splits: 78 total, 78 done (100.00%)
4.11 [9 rows, 5.62KB] [2 rows/s, 1.37KB/s]
```

one query **scan 10 times**?

2021-03-18T23:37:27.724+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:28.290+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:28.464+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:28.666+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:28.837+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:29.339+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:29.550+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:29.723+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:30.214+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true
2021-03-18T23:37:30.675+0800    INFO    Query-20210318_153723_00000_epdqw-324   org.apache.iceberg.TableScan    Scanning table odsx1.sample03 snapshot 9094447769635788533 created at 2021-03-18 23:05:50.986 with filter true

**Comment by electrum** [comment]
Note that we're not actually reading the data 10 times, but we are reading the metadata files 10 times, and since your table is tiny, reading the metadata will have significant overhead relative to the reading the data (and could even be more expensive than reading the data).

**Comment by rlcyf** [comment]
i testing on 353 now!
I am waiting for this issue https://github.com/trinodb/trino/issues/6982#event-4365772616  release to new version(354?)

**Comment by phd3** [comment]
Closed via #7367.
